forc/src/ops/forc_abi_json.rs

```diff
@@ -1,18 +1,30 @@
 use crate::cli::{BuildCommand, JsonAbiCommand};
 use anyhow::Result;
+use forc_pkg::check_program_type;
 use serde_json::{json, Value};
 use std::fs::File;
+use std::path::PathBuf;
+use sway_core::TreeType;
 
 pub fn build(command: JsonAbiCommand) -> Result<Value> {
+    let curr_dir = if let Some(ref path) = command.path {
+        PathBuf::from(path)
+    } else {
+        std::env::current_dir()?
+    };
+    check_program_type(curr_dir, TreeType::Contract)?;
+
     let build_command = BuildCommand {
         path: command.path,
         offline_mode: command.offline_mode,
         silent_mode: command.silent_mode,
         minify_json_abi: command.minify,
         ..Default::default()
     };
+
     let compiled = crate::ops::forc_build::build(build_command)?;
     let json_abi = json!(compiled.json_abi);
+
```

Comment:
Not important, but try to avoid adding newlines, touching whitespace or formatting unrelated code as it can cause a bit of git diff noise and occasionally cause unnecessary conflicts.

Don't worry, I'm guilty of accidentally removing trailing whitespace in random places because of my editor setup sometimes :face_with_spiral_eyes: